packages/token/src/tests/erc1155/test_fuzz_erc1155.cairo

```diff
@@ -0,0 +1,484 @@
+use crate::erc1155::ERC1155Component;
+use crate::erc1155::ERC1155Component::ERC1155CamelImpl;
+use crate::erc1155::ERC1155Component::{ERC1155Impl, ERC1155MetadataURIImpl, InternalImpl};
+use openzeppelin_introspection::src5::SRC5Component::SRC5Impl;
+use openzeppelin_test_common::erc1155::ERC1155SpyHelpers;
+use openzeppelin_test_common::erc1155::{deploy_another_account_at, setup_account, setup_receiver};
+use openzeppelin_test_common::mocks::erc1155::DualCaseERC1155Mock;
+use openzeppelin_testing::common::repeat;
+use openzeppelin_testing::constants::{BASE_URI, EMPTY_DATA, OPERATOR, RECIPIENT};
+
+use snforge_std::{spy_events, start_cheat_caller_address, test_address};
+use starknet::ContractAddress;
+
+//
+// Setup
+//
+
+type ComponentState = ERC1155Component::ComponentState<DualCaseERC1155Mock::ContractState>;
+
+fn COMPONENT_STATE() -> ComponentState {
+    ERC1155Component::component_state_for_testing()
+}
+
+const MIN_IDS_LEN: u32 = 2;
+const MAX_IDS_LEN: u32 = 10;
+
+const MIN_VALUE_MULT: u32 = 1;
+const MAX_VALUE_MULT: u32 = 1_000_000;
+
+#[derive(Copy, Drop)]
+struct TokenList {
+    ids: Span<u256>,
+    values: Span<u256>,
+}
+
+fn prepare_tokens(ids_len_seed: u32, value_mult_seed: u32) -> TokenList {
```

Comment:
We should be able to avoid this by using the new Fuzzing derivable trait from foundry 0.38.0. Let's update that before merging this to avoid the extra boilerplate.